server/src/main/java/io/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -43,6 +44,15 @@
    */
   List<DataSegment> getUsedSegmentsForInterval(String dataSource, Interval interval);
 
+  /**
+   * Get all used segments and the created_date of these segments in a given datasource and interval
+   * 
+   * @param dataSource The datasource to query
+   * @param interval   The interval for which all applicable and used datasources are requested. Start is inclusive, end is exclusive
+   * @return The DataSegments and the related created_date of segments which include data in the requested interval
+   */
+  List<Pair<DataSegment, String>> getUsedSegmentAndCreatedDateForInterval(String dataSource, Interval interval);
```

Comment:
I suggest to modify `List<DataSegment> getUsedSegmentsForInterval(String dataSource, Interval interval);` to return `List<Pair<DataSegment, String>>` rather than adding a new method.

Reply:
I don't know. I just think when someone calls method `getUsedSegmentsForInterval` , maybe he doesn't want to get the information about created date.

Reply:
Maybe created_date should be a part of `DataSegment`. In this way, we only need the method `List<DataSegment> getUsedSegmentsForInterval(String dataSource, Interval interval);`.  What do you think?

Reply:
The only usage of `getUsedSegmentsForInterval()` is `SegmentAllocateAction`. It checks any segments are already allocated for the given interval to allocate a new segment id. I think it can just ignore the createdDate part.

Reply:
> Maybe created_date should be a part of DataSegment. In this way, we only need the method List<DataSegment> getUsedSegmentsForInterval(String dataSource, Interval interval);. What do you think?

Hmm, that's a good point. It sounds good, but I'm not sure about why `created_date` is not a part of `DataSchema` itself. @gianm any idea?

Reply:
Alright, let me raise an issue for this and merge these two methods in another pr, because it affect about 16 Classes.

Reply:
Sounds good. Please go for it.